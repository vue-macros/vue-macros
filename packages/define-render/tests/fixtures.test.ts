import { resolve } from 'node:path'
import {
  rollupBuild,
  RollupVue,
  RollupVueJsx,
  testFixtures,
  UnpluginOxc,
} from '@vue-macros/test-utils'
import { describe } from 'vitest'
import VueDefineRender from '../src/rollup'

describe('fixtures', async () => {
  await testFixtures(
    'tests/fixtures/*.{vue,js,ts}',
    (args, id) =>
      rollupBuild(id, [
        RollupVue({
          isProduction: args.isProduction,
          inlineTemplate: args.isProduction,
        }),
        RollupVueJsx(),
        VueDefineRender(),
        UnpluginOxc.rollup(),
      ]),
    {
      cwd: resolve(__dirname, '..'),
      promise: true,
      params: [['isProduction', [true, false]]],
    },
  )
})
