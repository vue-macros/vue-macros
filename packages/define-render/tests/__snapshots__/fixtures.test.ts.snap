// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > tests/fixtures/basic.vue > isProduction = false 1`] = `
"// basic.js
import { defineComponent, h, createElementBlock, openBlock } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "basic",
	setup(__props, { expose: __expose }) {
		__expose();
		const __returned__ = {};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return () => h("div");
	}
});

function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", null, "basic");
}
var basic = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], __FILE__]);

export { basic as default };
"
`;

exports[`fixtures > tests/fixtures/basic.vue > isProduction = true 1`] = `
"// basic.js
import { defineComponent, h } from 'vue';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "basic",
	setup(__props) {
		return () => h("div");
	}
});

export { _sfc_main as default };
"
`;

exports[`fixtures > tests/fixtures/id.vue > isProduction = false 1`] = `
"// id.js
import { defineComponent, createVNode, createTextVNode } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "id",
	setup(__props, { expose: __expose }) {
		__expose();
		const renderFn = () => createVNode("div", null, [createTextVNode("Hello")]);
		const __returned__ = { renderFn };
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return renderFn;
	}
});

var id = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { id as default };
"
`;

exports[`fixtures > tests/fixtures/id.vue > isProduction = true 1`] = `
"// id.js
import { defineComponent, createVNode, createTextVNode } from 'vue';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "id",
	setup(__props) {
		const renderFn = () => createVNode("div", null, [createTextVNode("Hello")]);
		return renderFn;
	}
});

export { _sfc_main as default };
"
`;

exports[`fixtures > tests/fixtures/tsx.vue > isProduction = false 1`] = `
"// tsx.js
import { defineComponent, createVNode, createTextVNode, createElementBlock, openBlock } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "tsx",
	setup(__props, { expose: __expose }) {
		__expose();
		const __returned__ = {};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return () => createVNode("div", null, [createTextVNode("Hello")]);
	}
});

function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div");
}
var tsx = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], __FILE__]);

export { tsx as default };
"
`;

exports[`fixtures > tests/fixtures/tsx.vue > isProduction = true 1`] = `
"// tsx.js
import { defineComponent, createVNode, createTextVNode } from 'vue';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "tsx",
	setup(__props) {
		return () => createVNode("div", null, [createTextVNode("Hello")]);
	}
});

export { _sfc_main as default };
"
`;

exports[`fixtures > tests/fixtures/without-fn.vue > isProduction = false 1`] = `
"// without-fn.js
import { defineComponent, createVNode, createTextVNode } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "without-fn",
	setup(__props, { expose: __expose }) {
		__expose();
		const __returned__ = {};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return () => createVNode("div", null, [createTextVNode("Hello")]);
	}
});

var withoutFn = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { withoutFn as default };
"
`;

exports[`fixtures > tests/fixtures/without-fn.vue > isProduction = true 1`] = `
"// without-fn.js
import { defineComponent, createVNode, createTextVNode } from 'vue';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "without-fn",
	setup(__props) {
		return () => createVNode("div", null, [createTextVNode("Hello")]);
	}
});

export { _sfc_main as default };
"
`;
